### PR TITLE
Avoid lazy init exception in DefaultRegistryDataService 

### PR DESCRIPTION
**Pull request by agrancaric**
## Basic information

* nrich version:
2.1.1
* Module:
nrich-registry

## Description

Add entity initialization when returning a value from create or update method in DefaultRegistryDataService to avoid lazy init exception.

## Types of changes

- Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I have read the project's **CONTRIBUTING** document
- [x] I have checked my code with the project's static analysis tooling
- [x] I have formatted my code with the project's IDEA code-style configuration
- [x] I have checked my code for misspellings
- [x] I have organized my changes in easy-to-follow commits
- ~~My change requires a change to the documentation~~
- ~~I have updated the documentation accordingly~~
- [x] I have added tests to cover my changes
- [x] All new and existing tests pass.

### CODE REVIEW

File: nrich-registry/src/test/java/net/croz/nrich/registry/data/controller/RegistryDataControllerTest.java [hunk 1]
```diff
@@ -108,6 +108,21 @@ void shouldCreateRegistryEntity() throws Exception {
             .andExpect(jsonPath("$.name").value(entityName));
     }
 
+    @Test
+    void shouldCreateRegistryEntityWithAssociation() throws Exception {
```
Comment (jzrilic) [inline nrich-registry/src/test/java/net/croz/nrich/registry/data/controller/RegistryDataControllerTest.java:112]: Maybe `shouldCreateRegistryEntityAndReturnWithInitializedAssociation` or just `shouldCreateRegistryEntityWithInitializedAssociation` (if this is clear enough).

File: nrich-registry/src/main/java/net/croz/nrich/registry/data/util/HibernateUtil.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+import jakarta.persistence.Entity;
+import java.beans.PropertyDescriptor;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+public final class HibernateUtil {
```
Comment (jzrilic) [inline nrich-registry/src/main/java/net/croz/nrich/registry/data/util/HibernateUtil.java:33]: Maybe `HibernateInitializer` instead?

File: nrich-registry/src/test/java/net/croz/nrich/registry/data/util/testutil/HibernateUtilGeneratingUtil.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+    public static HibernateUtilTestEntity createAndSaveHibernateUtilTestEntity(EntityManager entityManager) {
+        HibernateUtilTestEntity firstChild = createHibernateUtilTestEntityInternal(entityManager, "first");
+        HibernateUtilTestEntity secondChild = createHibernateUtilTestEntityInternal(entityManager, "second");
+        HibernateUtilTestEntity parent = createHibernateUtilTestEntityInternal(entityManager, "parent");
+
+        return createHibernateUtilTestEntityInternal(entityManager, "main", parent, List.of(firstChild, secondChild));
+    }
+
+    private static HibernateUtilTestEntity createHibernateUtilTestEntityInternal(EntityManager entityManager, String name) {
```
Comment (jzrilic) [inline nrich-registry/src/test/java/net/croz/nrich/registry/data/util/testutil/HibernateUtilGeneratingUtil.java:38]: This method seems unnecessary. Just call the one below with two `null` arguments in the method above. It brings more confusion than benefit in my opinion.

File: nrich-registry/src/main/java/net/croz/nrich/registry/data/util/HibernateUtil.java (hunk 2 above)
Comment (agrancaric) [reply nrich-registry/src/main/java/net/croz/nrich/registry/data/util/HibernateUtil.java:33]: I like the name HibernateUtil better, besides we always call it like that, so maybe I would leave it.

**Review by jzrilic — APPROVED** [review]

**Pull request merged**